boards/arm/stm32h7b3i_dk/stm32h7b3i_dk.dts

```diff
@@ -62,17 +70,18 @@
 	status = "okay";
 };
 
+/* PLL1P is used for system clock (280 MHz), PLL1Q is used for FDCAN bit quantum clock (16 MHz) */
```

Comment:
16MHz seems a bit low. Does it pass all the CAN timing test? Have you tried with 80MHz (div-q = 7?) instead?

Reply:
Thanks for the quick review, this clock definitely needs changing. 
I have only tested on CAN2.0 compatible baudrates since I don't have a CANFD compatible adapter (I used PCAN-USB adapter and BUSMASTER on Windows).

Would testing with CAN controller in loopback mode (like in `samples/drivers/can` sample) be good enough?

Reply:
One little note about timing tests - I ran them with modified min timings in `can_stm32h7.c` because these assertions in `can_mcan.c` fail:
`
void can_mcan_configure_timing(struct can_mcan_reg *can,
			       const struct can_timing *timing,
			       const struct can_timing *timing_data)
{
	if (timing) {
		uint32_t nbtp_sjw = can->nbtp & CAN_MCAN_NBTP_NSJW_MSK;

		__ASSERT_NO_MSG(timing->prop_seg == 0);
		__ASSERT_NO_MSG(timing->phase_seg1 <= 0x100 &&
				timing->phase_seg1 > 0);
		__ASSERT_NO_MSG(timing->phase_seg2 <= 0x80 &&
				timing->phase_seg2 > 0);
		__ASSERT_NO_MSG(timing->prescaler <= 0x200 &&
				timing->prescaler > 0);
		__ASSERT_NO_MSG(timing->sjw == CAN_SJW_NO_CHANGE ||
				(timing->sjw <= 0x80 && timing->sjw > 0));

		can->nbtp = (((uint32_t)timing->phase_seg1 - 1UL) & 0xFF) <<
				CAN_MCAN_NBTP_NTSEG1_POS |
			    (((uint32_t)timing->phase_seg2 - 1UL) & 0x7F) <<
				CAN_MCAN_NBTP_NTSEG2_POS |
			    (((uint32_t)timing->prescaler  - 1UL) & 0x1FF) <<
				CAN_MCAN_NBTP_NBRP_POS;
`
This is because min timings config for `phase_seg1`, `phase_seg2` and `prescaler` in `can_stm32h7.c` are configured to `0x00`:
`
.timing_min = {
	.sjw = 0x00,
	.prop_seg = 0x00,
	.phase_seg1 = 0x00,
	.phase_seg2 = 0x00,
	.prescaler = 0x00
}
`
They should be configured to 0x01, since `can_mcan` decrements them by 1 before writing them to NBTP register.

Do I fix it in this PR or should I open another one?

Reply:
@henrikbrixandersen Thanks for the thorough review.
I added the fix for issue described above in a separate commit, please take a look if it's OK.

IMO, having `phase_seg1` and `phase_seg2` set to 0 doesn't make sense (where is the sampling point in that case?), it should be an invalid configuration (and it is rightfully checked in `can_mcan.c`). The driver first decrements the value by 1 and writes it to register, but hardware interprets the value as: value_in_register + 1.

Reply:
Indeed. Good catch. Please see my question about sjw min value.